modules/services/gpg-agent.nix

```diff
@@ -336,6 +354,7 @@ in
           ++ optional (cfg.maxCacheTtl != null) "max-cache-ttl ${toString cfg.maxCacheTtl}"
           ++ optional (cfg.maxCacheTtlSsh != null) "max-cache-ttl-ssh ${toString cfg.maxCacheTtlSsh}"
           ++ optional (cfg.pinentryPackage != null) "pinentry-program ${lib.getExe cfg.pinentryPackage}"
+          ++ optional (cfg.pinentryBinary != null) "pinentry-program ${cfg.pinentryBinary}"
```

Comment:
Wondering if, instead of supporting 2 different configurations for the same option, we just deprecate the package option in favor of supplying the program itself. 

Another idea, was that maybe we change it to be a `pinentry` option with sub options for `package` and `program` and `program` accepts a string to forward to `lib.getExe' cfg.pinentryPackage`.

Reply:
Thanks for the review -- No opinion here. If you tell me which one you like better, I'll do it.

Reply:
Created https://github.com/nix-community/home-manager/pull/6900, does that look like what you were hoping for in functionality?

Reply:
Nice. Let me pull that and test it out in a bit. Thanks!